src/rules/no-separate-loading-state.ts

```
@@ -0,0 +1,136 @@
+    schema: [],
+    messages: {
+      separateLoadingState: 'Avoid separate loading state. Encode loading status directly in the primary state using a sentinel value like "loading".',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    const loadingStateVariables = new Map<string, TSESTree.VariableDeclarator>();
+    const setterUsages = new Map<string, { truthy: boolean, falsy: boolean }>();
+
+    function isLoadingPattern(name: string): boolean {
+      return LOADING_PATTERNS.some(pattern => pattern.test(name));
```

ESLint

[error] Replace `pattern` with `(pattern)`

(prettier/prettier)